### DISCUSSION TITLE
Machine Duplicates Itself on the Web UI

### DISCUSSION THREAD

**Discussion by TheBlueKingLP** [Ylianst/MeshCentral#5728] [Issues] [answered]
**Describe the bug**

**To Reproduce**
1. Enable mac address privacy and configure it to switch mac address on each connect for a network adapter on a Linux machine
2. Install Mesh Central Agent
3. Disable Wi-Fi
4. Stop Mesh Central Agent
5. Reconnect to Wi-Fi
6. Start Mesh Central Agent
7. See duplicate instance of the machine

**Expected behavior**
Only 1 machine is shown and it will not get duplicated when doing the above step

**Screenshots**
![image](https://github.com/Ylianst/MeshCentral/assets/12997043/ab4e1f78-7783-449e-9132-9597c4e73fb4)

**Server Software (please complete the following information):**
 - OS: Photon OS
 - Virtualization: Docker
 - Network: WLAN (Laptop -> Site LAN -> Internet -> HQ LAN -> MeshCentral)
 - Version: 1.1.19
 - Node: 20.11.0

**Client Device (please complete the following information):**
 - Device: Laptop
 - OS: Arch Linux
 - Network: WLAN (Laptop -> Site LAN -> Internet -> HQ LAN -> MeshCentral)
 - Browser: Firefox Fork (LibreWolf)
 - MeshCentralRouter Version: Not Applicable

**Remote Device (please complete the following information):**
 - Device: Laptop
 - OS: Arch Linux
 - Network: Remote WLAN
 - Current Core Version (if known): Current Core: Nov 21 2022, 598124670

**Additional context**

**Your config.json file**
```
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "settings": {
    "cert": "amt.example.com",
    "sessionKey": "SessionKeyRedacted",
    "port": 443,
    "mpsport": 4433,
    "redirPort": 80,
    "AgentPong": 300,
    "TLSOffload": true,
    "SelfUpdate": false,
    "AllowFraming": "false",
    "mongodb": "mongodb://mongodb:27017/mesh",
    "mongodbcol": "mesh",
    "WebRTC": "false",
  },
  "domains": {
    "": {
      "minify": true,
      "NewAccounts": "false",
      "_userNameIsEmail": true,
      "certUrl": "https://amt.example.com",
      "AmtManager": {
        "WifiProfiles": [
          {
            "ssid": "SSID",
            "password": "PPASSWORDD",
            "encryption": "ccmp-aes",
            "authentication": "wpa2-psk"
          }
        ]
      }

    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before>",
    "_email": "myemail@mydomain.com",
    "_names": "myserver.mydomain.com",
    "production": false
  }
}

```

**Comment by si458** [comment, marked as answer]
This is not a bug but a question, this is expected behaviour...
The way the meshagent generates its meshid is based on the devices hardware and mac addresses, so if u change your interface mac addresses everything then you will start seeing duplicate entries of your machine

One way around this is to add `skipmaccheck=1` into the .msh file that's next to your exe then restart the agent,
This will basically tell the meshagent to not regenerate the meshid if the mac addresses change
But be aware this will have huge side effects if u use this option with cloned virtual machines!?

**Comment by trmdi** [comment]
https://github.com/Ylianst/MeshCentral/issues/5845
